common/src/main/scala/io/phdata/clients/HiveClient.scala

```diff
@@ -38,16 +38,20 @@ class HiveClientImpl[F[_]](loginContextProvider: LoginContextProvider, transacto
       location: String,
       comment: String,
       dbProperties: Map[String, String]
-  ): F[Unit] =
+  ): F[Unit] = {
+    val formattedLocation =
+      if (!location.startsWith("hdfs://") && !location.endsWith("/")) location ++ "/" else location
```

Comment:
You can add a comment saying that S3 requires a trailing slash so the path is recognized as a prefix